### ISSUE TITLE
cjxl incorrectly displays VarDCT as default for lossy (JPEG/GIF) input

### ISSUE THREAD

**Issue by jonnyawsom3** [libjxl/libjxl#3111]
**Describe the bug**
When using a GIF or Jpeg with transcoding disabled, the default behaviour of lossless encoding is shown as lossy instead.

**To Reproduce**
```
cjxl dancing-banana.gif dancing-banana.jxl
JPEG XL encoder v0.9.0 c51cffd [AVX2,SSE4,SSSE3,SSE2]
Encoding [VarDCT, d1.000, effort: 7]
Compressed to 6614 bytes (0.006 bpp/frame).
990 x 1050, 0.260 MP/s [0.26, 0.26], 1 reps, 16 threads.

cjxl dancing-banana.gif dancing-banana.jxl -d 1
JPEG XL encoder v0.9.0 c51cffd [AVX2,SSE4,SSSE3,SSE2]
Encoding [VarDCT, d1.000, effort: 7]
Compressed to 119.0 kB (0.114 bpp/frame).
990 x 1050, 0.097 MP/s [0.10, 0.10], 1 reps, 16 threads.

cjxl Test.jpg Test.jxl -j 0
JPEG XL encoder v0.9.0 c51cffd [AVX2,SSE4,SSSE3,SSE2]
Encoding [VarDCT, d1.000, effort: 7]
Compressed to 123.1 kB (2.347 bpp).
910 x 461, 0.796 MP/s [0.80, 0.80], 1 reps, 16 threads.

cjxl Test.jpg Test.jxl -j 0 -d 1
JPEG XL encoder v0.9.0 c51cffd [AVX2,SSE4,SSSE3,SSE2]
Encoding [VarDCT, d1.000, effort: 7]
Compressed to 28565 bytes (0.545 bpp).
910 x 461, 0.931 MP/s [0.93, 0.93], 1 reps, 16 threads.
```

**Expected behavior**
Default lossless encoding should display as Lossless Modular in the cjxl output.

The GIF used when I discovered the issue (Taken from Google's WebP example ironically).
![dancing-banana](https://github.com/libjxl/libjxl/assets/51539240/d1c57830-95bf-47ee-beb7-19221fdf9b73)

**Comment by sboukortt** [comment]
Thanks for catching this. It seems that the bug was introduced by #2948 and fixed by #3013.

@mo271 Let’s merge #3013 in v0.9.1?

**Comment by mo271** [comment]
cherry picking the fix for this for v0.9.1 in #3114
